### PR TITLE
Added docs for broadcast and UDF guide

### PR DESCRIPTION
**Pull request by Niharikadutta**
This PR adds the following two docs to the dotnet docs:

1. Broadcast guide - guide on how to use broadcast variables inside .NET for Apache Spark
2. UDF guide - guide on UDF serialization and caveats

Preview Links: 

File Type | File Name | Published Url
-- | -- | --
Content | docs/spark/how-to-guides/broadcast-guide.md | https://review.docs.microsoft.com/en-us/dotnet/spark/how-to-guides/broadcast-guide?branch=pr-en-us-18710
Content | docs/spark/how-to-guides/udf-guide.md | https://review.docs.microsoft.com/en-us/dotnet/spark/how-to-guides/udf-guide?branch=pr-en-us-18710

### CODE REVIEW

File: docs/spark/how-to-guides/udf-guide.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        <>c__DisplayClass0_1 <>c__DisplayClass0_2 = new <>c__DisplayClass0_1();
+        <>c__DisplayClass0_2.s2 = "s2";
+        Func<string, string> func2 = new Func<string, string>(<>c__DisplayClass0_2.<M>b__1);
+    }
+}
+```
+
+Notice that `func` and `func2` no longer share a closure, and they have their own separate closures `<>c__DisplayClass0_0` and `<>c__DisplayClass0_1` respectively. When used as the target for serialization, nothing other than the referenced variables will get serialized for the delegate. This behavior is important to keep in mind while implementing multiple UDFs in a common scope.
+
+To learn more about UDFs in general, please review the following articles that explain UDFs and how to use them: [UDFs in databricks(scala)](https://docs.databricks.com/spark/latest/spark-sql/udf-scala.html), [Spark UDFs and some gotchas](https://medium.com/@achilleus/spark-udfs-we-can-use-them-but-should-we-use-them-2c5a561fde6d).
```
Comment (luisquintanilla) [inline docs/spark/how-to-guides/udf-guide.md:178]: For the Spark UDF and gotchas, is there any  official documentation rather than a Medium post?

**Review by luisquintanilla — COMMENTED** [review]
Nicely done @Niharikadutta . I learned a thing or two :). Added a comment / question.

File: docs/spark/how-to-guides/udf-guide.md (hunk 1 above)
Comment (Niharikadutta) [reply docs/spark/how-to-guides/udf-guide.md:178]: I don't believe I was able to find an official documentation besides the one I linked at the start.

**Comment by Niharikadutta** [comment]
@luisquintanilla Thanks for reviewing! Please let me know if there are any changes needed or if this can be merged. Thanks!

File: docs/spark/how-to-guides/udf-guide.md (hunk 1 above)
Comment (luisquintanilla) [reply docs/spark/how-to-guides/udf-guide.md:178]: Hmmm...in that case, can we pull the reference to the Medium article. As an alternative, we can add a section at the bottom of this guide like the following:

## Spark UDF Caveats

- Null values in UDFs can throw exceptions. It's the responsibility of the developer to handle.
- UDFs don't leverage the optimizations provided by Spark's built-in functions.

**Comment by luisquintanilla** [comment]
@Niharikadutta Added another. Let me know if you have questions.

**Review by luisquintanilla — APPROVED** [review]
Looks great @Niharikadutta :shipit:

**Pull request merged**